hrms/hr/doctype/shift_assignment_tool/shift_assignment_tool.js

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Shift Assignment Tool", {
+	setup(frm) {
+		hrms.setup_employee_filter_group(frm);
+	},
+
+	async refresh(frm) {
+		frm.page.clear_indicator();
+		frm.disable_save();
```

Comment:
mandatory fields and employee selection should be validated on primary action click

<img width="1340" alt="image" src="https://github.com/frappe/hrms/assets/24353136/d5479a96-e4a1-4a83-83de-071ec116f18e">